packages/api/src/external/carequality/process-patient-discovery-progress.ts

```diff
@@ -18,13 +18,30 @@ export async function processPatientDiscoveryProgress({
   status,
 }: {
   patient: Pick<Patient, "id" | "cxId">;
-  status: "processing" | "completed" | "failed";
+  status: "processing" | "completed" | "failed" | "re-run";
 }): Promise<void> {
   const { log } = out(`CQ Process PD Status - patient ${patient.id}`);
 
   try {
     const updatedPatient = await updatePatientDiscoveryStatus({ patient, status });
 
+    if (status === "re-run") {
+      const requestId = updatedPatient.data.patientDiscovery?.requestId;
+      const facilityId = updatedPatient.facilityIds[0];
+      if (!requestId || !facilityId) {
+        log(
+          `Cannot trigger re-run of patient discovery with missing requestId or facilityId - patient ${patient.id}`
+        );
+      } else {
+        log(
+          `Triggering re-run of patient discovery with requestId ${requestId} and facilityId ${facilityId}`
+        );
+
+        discover(updatedPatient, facilityId, requestId, true);
+      }
+      return;
+    }
```

Comment:
I'd avoid adding this logic to this function, its increasing the amount of responsibilities, which makes it more complex.

Also, it's called in multiple places but only in one of those places/situations we'd consider running this.

Reply:
No longer relevant.